tools/testing/do_target_determination_for_s3.py

```diff
@@ -0,0 +1,82 @@
+import json
+import os
+import pathlib
+import sys
+
+
+REPO_ROOT = pathlib.Path(__file__).resolve().parent.parent.parent
+sys.path.insert(0, str(REPO_ROOT))
+
+from tools.stats.import_test_stats import (
+    copy_pytest_cache,
+    get_td_heuristic_historial_edited_files_json,
+    get_td_heuristic_profiling_json,
+    get_test_class_ratings,
+    get_test_class_times,
+    get_test_file_ratings,
+    get_test_times,
+)
+from tools.stats.upload_metrics import emit_metric
+
+from tools.testing.discover_tests import TESTS
+from tools.testing.target_determination.determinator import (
+    AggregatedHeuristics,
+    get_prediction_confidences,
+    get_test_prioritizations,
+    TestPrioritizations,
+)
+
+sys.path.remove(str(REPO_ROOT))
```

Comment:
Hmm, what does this one do?  As this is here, it'll be run whenever `from tools.testing.do_target_determination_for_s3 import import_results`  from `run_test` is run.  It makes me feel a bit uneasy seeing a script having write effect when someone imports it.

Reply:
This happens in run_test as well.  I believe that it's correct since the path modification is only needed for the import.
On the other hand, I'm pretty sure this means those files can't import torch